src/applications/personalization/profile/util/index.js

```diff
@@ -35,53 +60,146 @@ export async function getData(apiRoute, options) {
   }
 }
 
+// used for legacy, remove once moved to lighthouse endpoint
+const hasPPIUErrorText = (errors, errorKey, errorText) => {
+  return errors.some(err =>
+    err.meta?.messages?.some(
+      message =>
+        message.key === errorKey &&
+        message.text?.toLowerCase().includes(errorText.toLowerCase()),
+    ),
+  );
+};
+
+// used for legacy, remove once moved to lighthouse endpoint
+const hasPPIUErrorKey = (errors, errorKey) => {
+  return errors.some(err =>
+    err.meta?.messages?.some(message => message.key === errorKey),
+  );
+};
+
+const hasLighthouseErrorText = (errors, errorKey, errorText) => {
+  errors.some(
+    err =>
+      err?.code === errorKey &&
+      err?.detail?.toLowerCase().includes(errorText.toLowerCase()),
+  );
+};
+
+const hasLighthouseErrorKey = (errors, errorKey) =>
+  errors.some(err => err?.code === errorKey);
+
 const hasErrorMessage = (errors, errorKey, errorText) => {
   if (errorText) {
     return (
-      errors.some(err =>
-        err.meta?.messages?.some(
-          message =>
-            message.key === errorKey &&
-            message.text?.toLowerCase().includes(errorText.toLowerCase()),
-        ),
-      ) ||
-      errors.some(
-        err =>
-          err?.code === errorKey &&
-          err?.detail?.toLowerCase().includes(errorText.toLowerCase()),
-      )
+      hasPPIUErrorText(errors, errorKey, errorText) ||
+      hasLighthouseErrorText(errors, errorKey, errorText)
     );
   }
   return (
-    errors.some(err =>
-      err.meta?.messages?.some(message => message.key === errorKey),
-    ) || errors.some(err => err?.code === errorKey)
+    hasPPIUErrorKey(errors, errorKey) || hasLighthouseErrorKey(errors, errorKey)
   );
 };
 
 export const hasAccountFlaggedError = errors =>
-  hasErrorMessage(errors, ACCOUNT_FLAGGED_FOR_FRAUD_KEY);
+  hasErrorMessage(
+    errors,
+    PPIU_ERROR_MAP.ACCOUNT_FLAGGED_FOR_FRAUD.RESPONSE_KEY,
+  ) ||
+  hasErrorMessage(
+    errors,
+    LIGHTHOUSE_ERROR_KEYS.LH_ACCOUNT_FLAGGED_FOR_FRAUD_KEY,
+  );
```

Comment:
Each of these functions checks a specific key and|or a specific text being present in the error response. As you can also see, there is a check for PPIU and for LIGHTHOUSE keys, which will be refactored once the legacy endpoint is completely phased out. That will help makes these not look so verbose